packages/web3torrent/src/hooks/use-budget.ts

```diff
@@ -0,0 +1,48 @@
+import {web3torrent} from '../clients/web3torrent-client';
+import {INITIAL_BUDGET_AMOUNT} from '../constants';
+import {useState, useEffect} from 'react';
+import {SiteBudget} from '@statechannels/client-api-schema';
+import {useChannelClient} from './use-web3-torrent-client';
+
+export function useBudget() {
```

Comment:
This is a new pattern I'm looking at for accessing functionality of the clients. Instead of referencing properties off of the context we instead create hooks when needed to access different parts of the context.



We could also use something like [constate](https://github.com/diegohaz/constate) if we wanted with this pattern (but I'm not sure if I see the benefit of using the context over custom hooks)